### ISSUE TITLE
Hazelcast client outside AWS cannot connect to hazelcast node inside AWS : Regression

### ISSUE THREAD

**Issue by rajeshnair** [hazelcast/hazelcast#8944]
We have just upgraded from 3.3.4 to 3.6.2 
and looks like the below Client program broken

````  
      AwsConfig config = new AwsConfig();
        config.setAccessKey("***********");
        config.setSecretKey("*******");
        config.setHostHeader("ec2.us-east-1.amazonaws.com");
        config.setSecurityGroupName("*****");
        config.setTagKey("hazelcast_service");
        config.setTagValue("true");
        config.setEnabled(true);

         AWSClient aclient = new AWSClient(config);
        for(String s : aclient.getAddresses().keySet()) {
            System.out.print(s+ "  "+aclient.getAddresses().get(s));
        }
       

        ClientConfig clientConfig = new ClientConfig();
        ClientAwsConfig clientAwsConfig = new ClientAwsConfig();
        clientAwsConfig.setInsideAws( false );
        clientAwsConfig.setAccessKey(config.getAccessKey());
        clientAwsConfig.setSecretKey(config.getSecretKey());
        String region = config.getRegion();
        clientAwsConfig.setRegion(region);
        clientAwsConfig.setHostHeader( "ec2."+region+".amazonaws.com" );
        clientAwsConfig.setSecurityGroupName(config.getSecurityGroupName());
        clientAwsConfig.setEnabled(true);
        clientAwsConfig.setConnectionTimeoutSeconds(10000);

        clientConfig.getNetworkConfig().setAwsConfig(clientAwsConfig);
        clientConfig.getNetworkConfig().setConnectionTimeout(10000);

        String clusterName = "myhzcluster";
        clientConfig.getGroupConfig().setName(clusterName);
        String password = "hzdummypwd";
        clientConfig.getGroupConfig().setPassword(password);
        HazelcastClient.newHazelcastClient(clientConfig);
````

The same program connects fine with hazelcast 3.3.4 
but fails to connect with hazelcast 3.6.2 with following error

````
Exception in thread "main" java.lang.IllegalStateException: Unable to connect to any address in the config! The following addresses were tried:[/172.31.31.121:5703, localhost/127.0.0.1:5703, /172.31.31.121:5702, /172.31.31.121:5701, localhost/127.0.0.1:5702, localhost/127.0.0.1:5701]
	at com.hazelcast.client.spi.impl.ClusterListenerSupport.connectToCluster(ClusterListenerSupport.java:174)
	at com.hazelcast.client.spi.impl.ClientClusterServiceImpl.start(ClientClusterServiceImpl.java:196)
	at com.hazelcast.client.impl.HazelcastClientInstanceImpl.start(HazelcastClientInstanceImpl.java:307)
	at com.hazelcast.client.HazelcastClient.newHazelcastClient(HazelcastClient.java:86)
	at TestAWSClient.main(TestAWSClient.java:106)

````

It appears that it is trying to connect to private IP rather than public IP despite setInsideAws(false); .

**Comment by rajeshnair** [comment]
@bilalyasar 

Let me know if you want me to upload the Server and Client java program

**Comment by bilalyasar** [comment]
@rajeshbhardwaj, thanks. Configs/Logs are enough for now. I have reproduced the issue. I will try to send fix as soon as possible.

**Comment by bhushankh** [comment]
@bilalyasar  - Any update on when we can have the fix available. We need the fix in 3.6.2.

**Comment by rajeshnair** [comment]
@bilalyasar 

Is the below pull request final fix for this issue ? https://github.com/hazelcast/hazelcast/pull/8996/files

I do not see this in the master or 3.7.2 branch yet
Please let me know the right version in which this would be available

We have already requested this fix to be backported to 3.6.2 to Greg and Faud.
Do you see any issues with backporting?

**Comment by GirishKumar0310** [comment]
Hi , I also faced similar issue around AddressTranslator usage in 3.7.1. In fact our custom proxy stopped working. I had to make changes in two places in com.hazelcast.client.connection.nio.ClientConnectionManagerImpl to make it work

public ClientConnection getConnection(Address target) {
    **Address remoteAddress = addressTranslator.translate(target);
    if (remoteAddress == null) {
      return null;
    }**

And then in InitConnectionTask::run
try {
          Address remoteAddress = addressTranslator.translate(target);
          connection = createSocketConnection(remoteAddress);
        } catch (Exception e) {

Could you please let me know when we can expect the fix the for same? It is very critical for us as upgrade to 3.7.1 has broken the code
